MTFO/Custom/Scans.cs

```diff
@@ -9,32 +10,42 @@ public class ScanHolder
         public List<CustomClusterScan> Clusters { get; set; }
     }
 
+    public enum RevealDistanceMode
+    {
+        Distance,
+        Time
+    }
+
     public struct CustomBioScan
     {
         public uint BaseScan { get; set; }
         public uint PersistentID { get; set; }
-        public int PlayerRequirement { get; set; }
+        public PlayerRequirement PlayerRequirement { get; set; }
         public float ScanRadius { get; set; }
         public float[] PlayersInScanMulti { get; set; }
         public float ReduceSpeed { get; set; }
         public bool ReduceWhenNoPlayer { get; set; }
+        public float RevealTime { get; set; }
+        [JsonIgnore]
+        internal float SplineRevealSpeed => 1f / this.RevealTime;
+        public RevealDistanceMode RevealTimeDistanceMode { get; set; }
         public BioScanGx BioScanGraphics { get; set; }
 
 
         public struct BioScanGx
         {
             public string ScanText { get; set; }
             public float Radius { get; set; }
-            public BioScanColorByMode[] colorModeColor { get; set; }
+            public BioScanColorByMode[] ColorModeColor { get; set; }
         }
 
         public class BioScanColorByMode
         {
-            public eChainedPuzzleGraphicsColorMode mode { get; set; }
-            public float r { get; set; }
-            public float g { get; set; }
-            public float b { get; set; }
-            public float a { get; set; }
+            public eChainedPuzzleGraphicsColorMode Mode { get; set; }
+            public float R { get; set; }
+            public float G { get; set; }
+            public float B { get; set; }
+            public float A { get; set; }
```

Comment:
Wouldn't this break a lot of current configuration?

Reply:
No. Case-insensitive deserialization